### PR TITLE
OHRM5X-502: Employee WorkShift BE

### PR DESCRIPTION
**Pull request by KaveeshaSilva1996**

### CODE REVIEW

File: symfony/client/src/orangehrmLeavePlugin/components/EntitlementSituationalModal.vue [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <oxd-dialog
+    :gutters="false"
+    :style="{width: '90%', maxWidth: '500px'}"
```
Comment (RajithaKumara) [inline symfony/client/src/orangehrmLeavePlugin/components/EntitlementSituationalModal.vue:24]: can't we move inline styles to class

File: symfony/plugins/orangehrmLeavePlugin/test/Service/BasicWorkScheduleTest.php [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+use OrangeHRM\Pim\Service\EmployeeService;
+use OrangeHRM\Tests\Util\TestCase;
+
+/**
+ * @group Leave
+ * @group Service
+ */
+class BasicWorkScheduleTest extends TestCase
+{
+    public function testGetWorkShiftStartEndTime()
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmLeavePlugin/test/Service/BasicWorkScheduleTest.php:38]: `: void`

File: symfony/plugins/orangehrmLeavePlugin/test/Service/BasicWorkScheduleTest.php [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            ->method('getEmployeeService')
+            ->willReturn($employeeService);
+        $basicWorkSchedule->setEmpNumber(1);
+
+        $expectResult = new WorkShiftStartAndEndTime($startDate, $endDate);
+
+        $this->assertEquals($expectResult, $basicWorkSchedule->getWorkShiftStartEndTime());
+    }
+
+    public function testGetWorkShiftStartEndTimeEmployeeWorkShiftEmpty()
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmLeavePlugin/test/Service/BasicWorkScheduleTest.php:78]: `:void`

File: symfony/plugins/orangehrmLeavePlugin/test/Service/BasicWorkScheduleTest.php [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+            ->willReturn($defaultWorkShift);
+
+        $basicWorkSchedule = $this->getMockBuilder(BasicWorkSchedule::class)
+            ->onlyMethods(['getEmployeeService', 'getWorkShiftService'])
+            ->getMock();
+
+        $basicWorkSchedule->expects($this->exactly(1))
+            ->method('getEmployeeService')
+            ->willReturn($employeeService);
+        $basicWorkSchedule->expects($this->exactly(1))
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmLeavePlugin/test/Service/BasicWorkScheduleTest.php:118]: `$this->exactly(1)` ===> `$this->once()`

File: symfony/plugins/orangehrmPimPlugin/config/routes.yaml [hunk 5]
```diff
@@ -423,6 +423,16 @@ apiv2_pim_employees_us_tax_exemption:
   requirements:
     empNumber: '\d+'
 
+apiv2_pim_employees_work_shift:
+  path: /api/v2/pim/employees/{empNumber}/work-shift
+  controller: OrangeHRM\Core\Controller\Rest\V2\GenericRestController::handle
+  methods: [ GET ]
+  defaults:
+    id: 0
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/config/routes.yaml:431]: remove `id: 0`
define `_key: empNumber`

File: symfony/plugins/orangehrmPimPlugin/Api/EmployeeWorkShiftAPI.php [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/Api/EmployeeWorkShiftAPI.php:67]: remove this rule

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: symfony/plugins/orangehrmLeavePlugin/test/Service/BasicWorkScheduleTest.php (hunk 4 above)
Comment (KaveeshaSilva1996) [reply symfony/plugins/orangehrmLeavePlugin/test/Service/BasicWorkScheduleTest.php:118]: Changed all places

File: symfony/client/src/orangehrmLeavePlugin/components/EntitlementSituationalModal.vue (hunk 1 above)
Comment (KaveeshaSilva1996) [reply symfony/client/src/orangehrmLeavePlugin/components/EntitlementSituationalModal.vue]: Added to data part.

Comment (KaveeshaSilva1996) [reply symfony/client/src/orangehrmLeavePlugin/components/EntitlementSituationalModal.vue]: Added to the css file

File: symfony/client/src/orangehrmLeavePlugin/components/EntitlementSituationalModal.vue [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      </oxd-form-actions>
+    </div>
+  </oxd-dialog>
+</template>
+
+<script>
+import Dialog from '@orangehrm/oxd/core/components/Dialog/Dialog';
+
+export default {
+  name: 'leave-balance-modal',
```
Comment (Super-Chama) [inline symfony/client/src/orangehrmLeavePlugin/components/EntitlementSituationalModal.vue:52]: please add correct name

File: symfony/client/src/orangehrmLeavePlugin/pages/configure/holiday/SaveHoliday.vue [hunk 8]
```diff
@@ -123,6 +125,7 @@ export default {
 
   data() {
     return {
+      yearArray: [...yearRange(201)],
```
Comment (Super-Chama) [inline symfony/client/src/orangehrmLeavePlugin/pages/configure/holiday/SaveHoliday.vue:128]: unless the requirement specifies, try to use the same year range due to selecting year from calendar can be hard. (50 past years <- current year -> 50 future years)

Comment (KaveeshaSilva1996) [reply symfony/client/src/orangehrmLeavePlugin/pages/configure/holiday/SaveHoliday.vue:128]: As discussed with kavinda, there was a requirement to have date-range of 100 from both sides in the enterprise product.

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**